sale_start_end_dates/models/sale_order.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2014-2019 Akretion (http://www.akretion.com)
+# @author Alexis de Lattre <alexis.delattre@akretion.com>
+# Copyright 2016-2019 Sodexis (http://sodexis.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+from dateutil.relativedelta import relativedelta
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    default_start_date = fields.Date(string='Default Start Date')
+    default_end_date = fields.Date(string='Default End Date')
+
+    @api.constrains('default_start_date', 'default_end_date')
+    def _check_default_start_end_dates(self):
+        if (self.default_start_date and self.default_end_date and
+                self.default_start_date > self.default_end_date):
+            raise ValidationError(_(
+                "Default Start Date should be before or be the "
+                "same as Default End Date for sale order '%s'.")
+                % self.display_name)
+
+    @api.onchange('default_start_date')
+    def default_start_date_change(self):
+        if (self.default_start_date and self.default_end_date and
+                self.default_start_date > self.default_end_date):
+            self.default_end_date = self.default_start_date
+
+    @api.onchange('default_end_date')
+    def default_end_date_change(self):
+        if (self.default_start_date and self.default_end_date and
+                self.default_start_date > self.default_end_date):
+            self.default_start_date = self.default_end_date
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    start_date = fields.Date(
+        string='Start Date', readonly=True,
+        states={'draft': [('readonly', False)]})
+    end_date = fields.Date(
+        string='End Date', readonly=True,
+        states={'draft': [('readonly', False)]})
+    number_of_days = fields.Integer(
+        compute='_compute_number_of_days',
+        inverse='_inverse_number_of_days',
+        string='Number of Days', readonly=False, store=True)
+    must_have_dates = fields.Boolean(related='product_id.must_have_dates')
```

Comment:
As we are checking `must_have_dates` as multiple places (It will call the related function each time), Can't we make it `store=True`

Reply:
no benefit in the context of the migration of the module, in my opinion.